io.openems.edge.evcs.webasto.unite/src/io/openems/edge/evcs/webasto/unite/WebastoReadHandler.java

```diff
@@ -0,0 +1,62 @@
+package io.openems.edge.evcs.webasto.unite;
+
+import io.openems.edge.evcs.api.Status;
+
+// TODO: Can also be done by registering onSetNextValue listeners on the depending channel in the WebastoImpl.
+public class WebastoReadHandler {
+
+	private final WebastoImpl parent;
+
+	protected WebastoReadHandler(WebastoImpl parent) {
+		this.parent = parent;
+	}
+
+	protected void run() {
+		this.setPhaseCount();
+		this.setStatus();
+		this.parent._setChargePower((int) this.parent.getActivePower());
+	}
+
+	private void setStatus() {
+		switch (this.parent.getChargePointState()) {
+		case (0):
+			this.parent._setStatus(Status.NOT_READY_FOR_CHARGING);
+			break;
+		case (1):
+			this.parent._setStatus(Status.READY_FOR_CHARGING);
+			break;
+		case (2):
+			this.parent._setStatus(Status.CHARGING);
+			break;
+		case (3):
+		case (4):
+			this.parent._setStatus(Status.CHARGING_REJECTED);
+			break;
+		case (5):
+			// TODO Check if this state is also reached while paused
+			this.parent._setStatus(Status.CHARGING_FINISHED);
+			break;
+		case (7):
+		case (8):
+			this.parent._setStatus(Status.ERROR);
+		}
+	}
+
+	/**
+	 * Writes the Amount of Phases in the Phase channel.
+	 */
+	private void setPhaseCount() {
+		int phases = 0;
+
+		if (this.parent.getActivePowerL1() >= 100) {
```

Comment:
`100 W` seems rather arbitrary... ok, if it works. Maybe add a comment on why this is not `> 0`